src/main/kotlin/org/opensearch/commons/alerting/AlertingPluginInterface.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.commons.alerting
+
+import org.opensearch.action.ActionListener
+import org.opensearch.action.ActionResponse
+import org.opensearch.client.node.NodeClient
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.commons.alerting.action.AlertingActions
+import org.opensearch.commons.alerting.action.IndexMonitorRequest
+import org.opensearch.commons.alerting.action.IndexMonitorResponse
+import org.opensearch.commons.utils.BaseResponse
+import org.opensearch.commons.utils.recreateObject
+
+object AlertingPluginInterface {
```

Comment:
nit: Let's add Java doc with the intent that all the transport action interfaces for Alerting Plugin should be added here.

Reply:
added javadoc